include/clap/ext/draft/mini-curve-display.h

```diff
@@ -5,19 +5,87 @@
 // This extension allows a host to render a small curve provided by the plugin.
 // A useful application is to render an EQ frequency response in the DAW mixer view.
 
-static CLAP_CONSTEXPR const char CLAP_EXT_MINI_CURVE_DISPLAY[] = "clap.mini-curve-display/2";
+static CLAP_CONSTEXPR const char CLAP_EXT_MINI_CURVE_DISPLAY[] = "clap.mini-curve-display/3";
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+enum clap_mini_curve_display_curve_kind {
+   // The mini curve is intended to draw the total gain response of the plugin.
+   // In this case the y values are in dB and the x values are in hz (logarithmic).
+   // This would be useful in for example an equalizer.
+   CLAP_MINI_CURVE_DISPLAY_CURVE_KIND_GAIN_RESPONSE = 1,
+
+   // The mini curve is intended to draw the total phase response of the plugin.
+   // In this case the y values are in radians and the x values are in hz (logarithmic).
+   // This would be useful in for example an equalizer.
+   CLAP_MINI_CURVE_DISPLAY_CURVE_KIND_PHASE_RESPONSE = 2,
+
+   // The mini curve is intended to draw the transfer curve of the plugin.
+   // In this case the both x and y values are in dB.
+   // This would be useful in for example a compressor or distortion plugin.
+   CLAP_MINI_CURVE_DISPLAY_CURVE_KIND_TRANSFER_CURVE = 3,
+
+   // This mini curve is intended to draw gain reduction over time. In this case
+   // x refers to the window in seconds and y refers to level in dB, x_min is
+   // always 0, and x_max would be the duration of the window.
+   // This would be useful in for example a compressor or limiter.
+   CLAP_MINI_CURVE_DISPLAY_CURVE_KIND_GAIN_REDUCTION = 4,
+
+   // This curve is intended as a generic time series plot. In this case
+   // x refers to the window in seconds. x_min is always 0, and x_max would be the duration of the
+   // window.
+   // Y is not specified and up to the plugin.
+   CLAP_MINI_CURVE_DISPLAY_CURVE_KIND_TIME_SERIES = 5,
+
+   // Note: more entries could be added here in the future
+};
+
+typedef struct clap_mini_display_curve_hints {
+
+   // Range for the x axis.
+   double x_min;
+   double x_max;
+
+   // Range for the y axis.
+   double y_min;
+   double y_max;
+
+} clap_mini_display_curve_hints_t;
+
+// A set of points representing the curve to be painted.
+typedef struct clap_mini_display_curve_data {
+   // Indicates the kind of curve those values represent, the host can use this
+   // information to paint the curve using a meaningful color.
+   int32_t curve_kind;
+
+   // values[0] will be the leftmost value and values[data_size -1] will be the rightmost
+   // value.
+   //
+   // The value 0 and UINT16_MAX won't be painted.
+   // The value 1 will be at the bottom of the curve and UINT16_MAX - 1 will be at the top.
```

Comment:
What is the motivation for this? Ignoring 0 seems like a strange complication for plugins and hosts to work around.

Reply:
It is useful to hide some parts of the curve.
If you have an EQ with an high pass, you may not want to paint the entire rect.